pkg/autoscaler/autoscaler.go

```diff
@@ -259,7 +260,103 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 
 			uniqueJobName := fmt.Sprintf("%s-scale-from-zero", workloadJobName)
 			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s", uniqueJobName, expectedReplicas, workloadMemRequest.String()))
-			workload := framework.NewWorkLoad(expectedReplicas, workloadMemRequest, uniqueJobName, autoscalingTestLabel, targetedNodeLabel, "")
+			workload := framework.NewWorkLoad(expectedReplicas, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "", corev1.NodeSelectorRequirement{
+				Key:      targetedNodeLabel,
+				Operator: corev1.NodeSelectorOpExists,
+			})
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", workloadJobName)
+
+			Eventually(func() bool {
+				ms, err := framework.GetMachineSet(ctx, client, machineSet.GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet %s", machineSet.GetName())
+
+				By(fmt.Sprintf("Waiting for machineSet replicas to scale out. Current replicas are %v, expected %v.",
+					*ms.Spec.Replicas, expectedReplicas))
+
+				return *ms.Spec.Replicas == expectedReplicas
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "MachineSet %s failed to scale out to %d replicas", machineSet.GetName(), expectedReplicas)
+
+			By("Waiting for the machineSet replicas to become nodes")
+			framework.WaitForMachineSet(ctx, client, machineSet.GetName())
+
+			expectedReplicas = 0
+			By("Deleting the workload")
+			Expect(deleteObject(workload.Name, cleanupObjects[workload.Name])).Should(Succeed(), "Failed to delete scale-out workload %s", workload.Name)
+			delete(cleanupObjects, workload.Name)
+			Eventually(func() bool {
+				ms, err := framework.GetMachineSet(ctx, client, machineSet.GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet %s", machineSet.GetName())
+
+				By(fmt.Sprintf("Waiting for machineSet replicas to scale in. Current replicas are %v, expected %v.",
+					*ms.Spec.Replicas, expectedReplicas))
+
+				return *ms.Spec.Replicas == expectedReplicas
+			}, framework.WaitLong, pollingInterval).Should(BeTrue(), "MachineSet %s failed to scale in to 0 replicas", machineSet.GetName())
+		})
+
+		// Machines required for test: 1
+		// Reason: This tests checks that autoscaler is able to scale from zero when a workload requires specific architecture in the node affinity fields.
+		// Moreover, this test gives a better signal when multiple architectures are available in the cluster or the cluster is not amd64,
+		// and the workload requires to be scheduled on an architecture different from amd64.
+		It("It scales from/to zero a machine set with the architecture requested by the workload", func() {
```

Comment:
i wonder if this test should create 2 machinesets, one with the desired arch, and one without the desired arch, so that we can be sure the autoscaler chooses to scale the proper machineset?

Reply:
If we create two machine sets, the test will only run on multi-arch compute clusters using the multi-arch payload. I wanted to avoid it as we don't run this suite against heterogeneous clusters in the periodic jobs.

By creating only one, if no heterogeneous-in-architecture MachineSets are available in the cluster,  the test can run and has a good validity (better when tested on non-amd64 single-arch clusters): we set the nodeSelectorRequirement for the kubernetes.io/arch label in the workload and ensure that the autoscaler:
1. (non-amd64) doesn't use the default amd64 label set at https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go#L368-L378
2. ensure that the autoscaler considers the `kubernetes.io/arch` label when [checking the nodeaffinity predicates](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/simulator/predicatechecker/schedulerbased.go#L139) and choose the node group to scale

Reply:
Alternatively, we can create N machineset as the number of heterogeneous-in-arch machinesets already defined in the cluster. In the case of single-arch clusters, it would be 1; in the case of x86_64+aarch64 clusters, 2.

Reply:
i don't want to complicate the testing configuration, but at some point in the future it might be nice to have the ability to test with multiple machinesets.

maybe an aspirational `TODO` comment here would be good, saying that in the future if it is easy to create multi-arch clusters then we should consider the idea of having 2 machinesets (1 with the proper arch for the workload and 1 without).

Reply:
I finally implemented a tentative test with multiple machine sets (when possible).

Here is an excerpt of the run. In a dirty cluster, it took ~5 mins more than the last run I added in the comments because of (1) waiting for the cluster autoscaler to become leader (~3mins), (2) waiting for the jobs t complete before continuing with the scale-in (~1min).

I didn't squash the commits yet, so that we can go back to the previous implementation and leave it as it was yesterday if we don't want that.


```
Running Suite: Machine Suite - /var/home/aleskandro2/__hot_repos/cluster-api-actuator-pkg/pkg
=============================================================================================
Random Seed: 1689089548

Will run 1 of 38 specs
------------------------------
[BeforeSuite] 
/var/home/aleskandro2/__hot_repos/cluster-api-actuator-pkg/pkg/e2e_test.go:47
[BeforeSuite] PASSED [0.624 seconds]
------------------------------
SSSSSSSSSS
------------------------------
Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count It scales from/to zero a machine set with the architecture requested by the workload [autoscaler, periodic]
/var/home/aleskandro2/__hot_repos/cluster-api-actuator-pkg/pkg/autoscaler/autoscaler.go:303
I0711 17:32:34.133552   55445 autoscaler.go:141] Allocatable memory capacity of worker node "ip-10-0-145-154.us-east-2.compute.internal" is 14837092Ki
  STEP: Creating ClusterAutoscaler @ 07/11/23 17:32:34.135
  STEP: Starting Cluster Autoscaler event watcher @ 07/11/23 17:32:34.429
I0711 17:32:35.621448   55445 autoscaler.go:311] Platform is AWS
  STEP: Creating a new MachineSet with 0 replicas for each machineset of a different architecture found in the cluster @ 07/11/23 17:32:35.621
I0711 17:32:36.214231   55445 nodes.go:102] MachineSet "adistefa-a1-wb9bf-worker-us-east-2a" have 1 nodes
I0711 17:32:36.805516   55445 nodes.go:102] MachineSet "adistefa-a1-wb9bf-worker-us-east-2a-additional" have 2 nodes
I0711 17:32:37.250351   55445 nodes.go:102] MachineSet "adistefa-a1-wb9bf-worker-us-east-2b" have 1 nodes
I0711 17:32:37.550408   55445 nodes.go:102] MachineSet "adistefa-a1-wb9bf-worker-us-east-2c" have 1 nodes
  STEP: Deploying the MachineSet with 0 replicas (architecture: amd64) @ 07/11/23 17:32:37.55
  STEP: Creating a MachineAutoscaler backed by MachineSet openshift-machine-api/adistefa-a1-wb9bfhn8vt - min:0, max:1 @ 07/11/23 17:32:38.014
  STEP: Deploying the MachineSet with 0 replicas (architecture: arm64) @ 07/11/23 17:32:38.306
  STEP: Creating a MachineAutoscaler backed by MachineSet openshift-machine-api/adistefa-a1-wb9bfqzx9k - min:0, max:1 @ 07/11/23 17:32:38.762
  STEP: Creating scale-out workload e2e-autoscaler-workload-scale-from-zero: jobs: 2, memory: 1.0635227e+10 @ 07/11/23 17:32:38.909
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:39.343
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:42.491
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:45.639
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:48.785
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:51.932
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:55.082
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:32:58.228
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:01.374
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:04.519
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:07.67
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:10.817
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:13.962
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:17.109
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:20.257
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:23.403
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:26.55
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:29.696
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:32.844
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:35.992
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:39.137
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:42.283
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:45.432
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:48.578
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:51.726
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:54.872
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:33:58.021
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:01.168
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:04.314
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:07.461
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:10.609
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:13.755
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:16.902
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:20.048
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:23.195
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:26.344
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:29.49
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:32.636
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:35.785
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:38.932
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:42.08
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:45.233
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:48.379
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:51.527
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:54.674
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:34:57.82
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:00.968
I0711 17:35:02.679217   55445 autoscaler.go:198] cluster-autoscaler: cluster-autoscaler-default-64d5b95d48-tt22c became leader
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:04.128
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:07.276
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:10.425
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:13.571
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:16.719
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:19.87
I0711 17:35:20.393075   55445 autoscaler.go:198] cluster-autoscaler-status: Scale-up: setting group MachineSet/openshift-machine-api/adistefa-a1-wb9bfqzx9k size to 1 instead of 0 (max: 1)
I0711 17:35:20.995435   55445 autoscaler.go:198] cluster-autoscaler-status: Scale-up: group MachineSet/openshift-machine-api/adistefa-a1-wb9bfqzx9k size set to 1 instead of 0 (max: 1)
I0711 17:35:21.790654   55445 autoscaler.go:198] e2e-autoscaler-workload-scale-from-zero-spvhr: pod triggered scale-up: [{MachineSet/openshift-machine-api/adistefa-a1-wb9bfqzx9k 0->1 (max: 1)}]
I0711 17:35:21.795384   55445 autoscaler.go:198] e2e-autoscaler-workload-scale-from-zero-vbl58: pod triggered scale-up: [{MachineSet/openshift-machine-api/adistefa-a1-wb9bfqzx9k 0->1 (max: 1)}]
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:23.017
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:26.163
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:29.31
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:32.459
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:35.607
I0711 17:35:36.608122   55445 autoscaler.go:198] e2e-autoscaler-workload-scale-from-zero-vbl58: pod didn't trigger scale-up: 1 node(s) didn't match Pod's node affinity/selector, 1 max node group size reached
  STEP: Current replicas for adistefa-a1-wb9bfhn8vt are 0, expected 0. @ 07/11/23 17:35:38.753
  STEP: Waiting for machineSet replicas to scale out. Current replicas are 1, expected 1. @ 07/11/23 17:35:39.343
  STEP: Waiting for the machineSet replicas to become nodes @ 07/11/23 17:35:39.343
I0711 17:38:44.400012   55445 autoscaler.go:198] e2e-autoscaler-workload-scale-from-zero-spvhr: pod didn't trigger scale-up: 1 node(s) didn't match Pod's node affinity/selector, 1 max node group size reached
I0711 17:40:49.293535   55445 autoscaler.go:198] e2e-autoscaler-workload-scale-from-zero-vbl58: pod didn't trigger scale-up: 1 max node group size reached, 1 node(s) didn't match Pod's node affinity/selector
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:33.251
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:36.395
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:39.54
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:42.686
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:45.831
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:48.975
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:52.12
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:55.266
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:44:58.412
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:45:01.558
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:45:04.703
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:45:07.85
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:45:10.996
  STEP: Waiting for the workload to complete. Current completions are 0, expected 2. @ 07/11/23 17:45:14.141
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:17.285
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:20.433
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:23.579
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:26.725
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:29.87
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:33.018
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:36.165
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:39.311
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:42.457
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:45.604
  STEP: Waiting for the workload to complete. Current completions are 1, expected 2. @ 07/11/23 17:45:48.75
  STEP: Waiting for the workload to complete. Current completions are 2, expected 2. @ 07/11/23 17:45:51.897
  STEP: Deleting the workload @ 07/11/23 17:45:51.897
I0711 17:45:51.897391   55445 autoscaler.go:122] [cleanup] "e2e-autoscaler-workload-scale-from-zero" (*v1.Job)
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:45:52.191
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:45:55.337
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:45:58.484
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:01.629
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:04.774
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:07.922
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:11.07
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:14.215
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:17.364
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:20.512
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:23.657
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:26.805
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:29.954
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:33.101
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:36.249
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:39.395
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:42.542
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:45.69
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:48.835
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:51.983
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:55.13
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:46:58.277
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:47:01.423
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:47:04.569
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:47:07.718
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:47:10.865
I0711 17:47:11.658130   55445 autoscaler.go:198] ip-10-0-158-182.us-east-2.compute.internal: marked the node as toBeDeleted/unschedulable
I0711 17:47:11.664450   55445 autoscaler.go:198] cluster-autoscaler-status: Scale-down: removing empty node "ip-10-0-158-182.us-east-2.compute.internal"
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 1, expected 0. @ 07/11/23 17:47:14.013
I0711 17:47:16.649767   55445 autoscaler.go:198] dns-default-wd27b: deleting pod for node scale down
I0711 17:47:16.709352   55445 autoscaler.go:198] ip-10-0-158-182.us-east-2.compute.internal: nodes removed by cluster autoscaler
I0711 17:47:16.716931   55445 autoscaler.go:198] cluster-autoscaler-status: Scale-down: empty node ip-10-0-158-182.us-east-2.compute.internal removed
  STEP: Waiting for machineSet replicas to scale in. Current replicas are 0, expected 0. @ 07/11/23 17:47:17.158
  STEP: Stopping Cluster Autoscaler event watcher @ 07/11/23 17:47:17.158
  STEP: Waiting for ClusterAutoscaler to delete. @ 07/11/23 17:47:17.158
I0711 17:47:17.158683   55445 autoscaler.go:122] [cleanup] "default" (*v1.ClusterAutoscaler)
I0711 17:47:20.602022   55445 autoscaler.go:122] [cleanup] "adistefa-a1-wb9bfhn8vt" (*v1beta1.MachineSet)
I0711 17:47:20.753245   55445 autoscaler.go:122] [cleanup] "autoscale-adistefa-a1-wb9bfhn8vt4pxjv" (*v1beta1.MachineAutoscaler)
I0711 17:47:20.901789   55445 autoscaler.go:122] [cleanup] "adistefa-a1-wb9bfqzx9k" (*v1beta1.MachineSet)
I0711 17:47:21.050452   55445 autoscaler.go:122] [cleanup] "autoscale-adistefa-a1-wb9bfqzx9k2k7wv" (*v1beta1.MachineAutoscaler)
  STEP: Waiting for MachineSets to be deleted... @ 07/11/23 17:47:21.2
• [919.297 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.004 seconds]
------------------------------

Ran 1 of 38 Specs in 919.923 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 37 Skipped
PASS

Ginkgo ran 1 suite in 15m24.788426938s
Test Suite Passed
```